istioctl/cmd/injector-list.go

```diff
@@ -0,0 +1,324 @@
+		},
+	}
+
+	cmd.AddCommand(injectorListCommand())
+	return cmd
+}
+
+func injectorListCommand() *cobra.Command {
+	var opts clioptions.ControlPlaneOptions
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List sidecar injector and sidecar versions",
+		Long:    `List sidecar injector and sidecar versions`,
+		Example: `  istioctl experimental injector list`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := kubeClientWithRevision(kubeconfig, configContext, opts.Revision)
+			if err != nil {
+				return fmt.Errorf("failed to create k8s client: %v", err)
+			}
+			ctx := context.Background()
+
+			nslist, err := getNamespaces(ctx, client)
+			if err != nil {
+				return err
+			}
+			sort.Slice(nslist, func(i, j int) bool {
+				return nslist[i].Name < nslist[j].Name
+			})
+
+			hooks, err := getWebhooks(ctx, client)
+			if err != nil {
+				return err
+			}
+			pods, err := getPods(ctx, client)
+			if err != nil {
+				return err
+			}
+			err = printNS(cmd.OutOrStdout(), nslist, hooks, pods)
+			if err != nil {
+				return err
+			}
+			cmd.Println()
+			injectedImages, err := getInjectedImages(ctx, client)
+			if err != nil {
+				return err
+			}
+
+			sort.Slice(hooks, func(i, j int) bool {
+				return hooks[i].Name < hooks[j].Name
+			})
+			return printHooks(cmd.OutOrStdout(), nslist, hooks, injectedImages)
+		},
+	}
+
+	return cmd
+}
+
+func getNamespaces(ctx context.Context, client kube.ExtendedClient) ([]v1.Namespace, error) {
+	nslist, err := client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return []v1.Namespace{}, err
+	}
+	return nslist.Items, nil
+}
+
+func printNS(writer io.Writer, namespaces []v1.Namespace, hooks []admit_v1.MutatingWebhookConfiguration,
+	allPods map[resource.Namespace][]v1.Pod) error {
+
+	outputCount := 0
+
+	w := new(tabwriter.Writer).Init(writer, 0, 8, 1, ' ', 0)
+	for _, namespace := range namespaces {
+		if hideFromOutput(resource.Namespace(namespace.Name)) {
+			continue
+		}
+
+		revision := getInjectedRevision(&namespace, hooks)
+		podCount := podCountByRevision(allPods[resource.Namespace(namespace.Name)], revision)
+		if len(podCount) == 0 {
+			// This namespace has no pods, but we wish to display it if new pods will be auto-injected
+			if revision != "" {
+				podCount[revision] = revisionCount{}
+			}
+		}
+		for injectedRevision, count := range podCount {
+			if outputCount == 0 {
+				fmt.Fprintln(w, "NAMESPACE\tISTIO-REVISION\tPOD-REVISIONS")
+			}
+			outputCount++
+
+			fmt.Fprintf(w, "%s\t%s\t%s\n", namespace.Name, revision, renderCounts(injectedRevision, count))
+		}
+	}
+	if outputCount == 0 {
+		fmt.Fprintf(writer, "No Istio injected namespaces present.\n")
+	}
+
+	return w.Flush()
+}
+
+func getWebhooks(ctx context.Context, client kube.ExtendedClient) ([]admit_v1.MutatingWebhookConfiguration, error) {
+	hooks, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().List(ctx, metav1.ListOptions{})
```

Comment:
seems we need a filter here in case there are other MutatingWebhookConfigurations in the cluster?